homeassistant/components/control4/__init__.py

```diff
@@ -0,0 +1,221 @@
+"""The Control4 integration."""
+import asyncio
+import json
+import logging
+
+from aiohttp import client_exceptions
+from pyControl4.account import C4Account
+from pyControl4.director import C4Director
+from pyControl4.error_handling import BadCredentials
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_PASSWORD,
+    CONF_SCAN_INTERVAL,
+    CONF_TOKEN,
+    CONF_USERNAME,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers import device_registry as dr, entity
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import (
+    CONF_ACCOUNT,
+    CONF_CONFIG_LISTENER,
+    CONF_CONTROLLER_UNIQUE_ID,
+    CONF_DIRECTOR,
+    CONF_DIRECTOR_ALL_ITEMS,
+    CONF_DIRECTOR_MODEL,
+    CONF_DIRECTOR_SW_VERSION,
+    CONF_DIRECTOR_TOKEN_EXPIRATION,
+    DEFAULT_SCAN_INTERVAL,
+    DOMAIN,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS = ["light"]
+
+
+async def async_setup(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Stub to allow setting up this component.
+
+    Configuration through YAML is not supported at this time.
+    """
+    hass.data.setdefault(DOMAIN, {})
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up Control4 from a config entry."""
+    entry_data = hass.data[DOMAIN].setdefault(entry.entry_id, {})
+
+    config = entry.data
+    account = C4Account(config[CONF_USERNAME], config[CONF_PASSWORD])
```

Comment:
Since the library is using aiohttp, we recommend it to accept an optional aiohttp client session argument. We want all integrations that use aiohttp to use the home assistant aiohttp client session. It can be retrieved using `async_get_clientsession` in the `aiohttp_client` helper module.
https://github.com/home-assistant/core/blob/8beaccf2ddb5a6e67619738996b86420b7ba831e/homeassistant/helpers/aiohttp_client.py#L32-L36

Reply:
Is it necessary to implement this in `config_flow.py` as well? The `HomeAssistant` object doesn't seem to be passed into any `config_entries.ConfigFlow` functions, so I'm not sure how I would get the client session when validating config.

Reply:
@philhawthorne `self.hass` should be set in the config flow

Reply:
`self.hass`

Reply:
oops my comment was on an open tab that was no longer updated with latest comments 😬

Reply:
Thanks for the advice everyone 😅 I've added this to `config_flow.py` now.